runtime/tutor.txt

```diff
@@ -415,6 +422,303 @@ _________________________________________________________________
+   the current selection.
+
+ * Press semicolon ( ; ) to collapse
+
+ * Press f / F to extend selection up to & including a character.
+
+ * press t / T to extend selection until a character.
+
+
+
+
+
+=================================================================
+=                   COPYING AND PASTING TEXT                    =
+=================================================================
+
+ Press y to yank (copy) the selection to the clipboard.
+ Press p to paste the yanked selection after the cursor.
+ Press uppercase P to paste the yanked selection *before*
+ the cursor.
+
+ 1. Move the cursor to the line below marked -->.
+ 2. press w to select "banana" and y to yank it.
+ 3. move between "2" and "3" and press p to paste.
+ 4. Repeat between "3" and "4"
+
+
+ --> 1 banana 2 3 4
+
+ Note: Whenever you delete (d) or change (c) text, helix will
+       copy the deleted/changed text to the clipboard, you can
+       avoid this with alt-d and alt-c respectively.
+ Note: Helix doesn't share the system clipboard by default.
+       Press space-Y and space-P to yank/paste to/from
+       your computers main clipboard
+
+
+
+
+
+=================================================================
+=                        SWITCHING CASE                         =
+=================================================================
+
+ Press ~ to switch the case (capitalised-ness) of all selected
+ letters - or just the one under the cursor.
+
+ 1. Move the cursor to the line below marked -->.
+ 2. Select each wrongly capitalised or un-capitalised letter
+    and press ~ over them
+ (remember, the first letter *should* be capitalised)
+
+ --> thIs sENtencE hAs MIS-cApitalIsed leTTerS.
+
+
+
+
+
+=================================================================
+=                         SETTING CASE                          =
+=================================================================
+
+ Press ` (above tab key on most keyboards) to set all selected
+ letters to lowercase.
+
+ 1. Move the cursor to the line below marked -->.
+ 2. press x to select the line.
+ 3. press ` to change it to all lowercase
+
+--> this SENTENCE SHOULD all be in LOWERCASE!
+
+ Note: Conversely, alt-` changes all selected letter to uppercase.
+
+
+
+
+
+=================================================================
+=                            MACROS                             =
+=================================================================
+
+ Macros are just a way to record a set of actions that you want
+ to repeat.
+
+ Press Q to start recording a macro, you should see a poup at
+ the bottom of your screen. Press Q again to stop recording.
+ Press q to repeat your recorded macro.
+
+ 1. Move the cursor to the first line below marked -->.
+ 2. Press Q to start recording.
+ 3. Enter insert mode and type some text.
+ 4. Exit insert and press Q again to stop recording
+ 5. Move to the line below and press q to repeat the macro.
+
+ -->
+ -->
```

Comment:
This doesn't really fully encapsulate the extend of macros' capabilities. It should at least demonstrate the fact that you can record other editing actions.